
<launch>
   
   <!-- RGB-D LOCALIZATION VERSION -->
     
  <!-- TF FRAMES -->
  <node pkg="tf" type="static_transform_publisher" name="base_to_camera_tf" 
    args="0.0 0.0 0.0 0.0 0.0 0.0 /base_link /camera_link 100" />
     
  <group ns="rtabmap">
  
    <!-- Odometry -->
    <node pkg="rtabmap" type="visual_odometry" name="visual_odometry" output="screen">
      <remap from="rgb/image"       to="/camera/rgb/image_rect_color"/>
      <remap from="depth/image"     to="/camera/depth_registered/image_raw"/>
      <remap from="rgb/camera_info" to="/camera/depth_registered/camera_info"/>
   
      <param name="frame_id" type="string" value="base_link"/>
    
      <!-- RTAB-Map's parameters: do "rosrun rtabmap visual_odometry (double-dash)params" to see the list of available parameters. -->
      <param name="Odom/Type" type="string" value="1"/>            <!-- 0=SURF/SIFT 1=FAST/BRIEF -->
      <param name="Odom/ResetCountdown" type="string" value="0"/>  <!-- 0=disabled, otherwise countdown before auto-resetting odometry when lost. -->
    </node>
  
    <!-- Visual SLAM (robot side) -->
    <!-- args: "delete_db_on_start" and "udebug" -->
    <node name="rtabmap" pkg="rtabmap" type="rtabmap" output="screen" args="">
      <param name="subscribe_depth" type="bool" value="true"/>
      <param name="subscribe_laserScan" type="bool" value="false"/>
	
      <remap from="rgb/image" to="/camera/rgb/image_rect_color"/>
      <remap from="depth/image" to="/camera/depth_registered/image_raw"/>
      <remap from="rgb/camera_info" to="/camera/depth_registered/camera_info"/>
      <remap from="odom" to="odom"/>
	
      <param name="queue_size" type="int" value="30"/>
      
      <!-- RTAB-Map's parameters: do "rosrun rtabmap rtabmap (double-dash)params" to see the list of available parameters. -->
      <param name="Rtabmap/DatabasePath" type="string" value="~/.ros/rtabmap.db"/>  <!-- Database used for localization -->
      <param name="Rtabmap/DetectionRate" type="string" value="1"/>     <!-- Don't need to do relocation very often! Though better results if the same rate as when mapping. -->
      <param name="Mem/STMSize" type="string" value="1"/>               <!-- 1 location in short-term memory -->
      <param name="Mem/IncrementalMemory" type="string" value="false"/>  <!-- false = Localization mode-->
      <param name="LccBow/MinInliers" type="string" value="10"/>        <!-- Minimum inliers to accept a loop closure -->
    </node>
  
    <!-- Visualisation (client side) -->
    <node pkg="rtabmap" type="rtabmapviz" name="rtabmapviz" args="-d $(find rtabmap)/launch/config/rgbd_gui.ini" output="screen">
      <param name="subscribe_depth" type="bool" value="true"/>
      <param name="subscribe_laserScan" type="bool" value="false"/>
      <param name="queue_size" type="int" value="30"/>
    
      <remap from="rgb/image" to="/camera/rgb/image_rect_color"/>
      <remap from="depth/image" to="/camera/depth_registered/image_raw"/>
      <remap from="rgb/camera_info" to="/camera/depth_registered/camera_info"/>
      <remap from="odom" to="odom"/>
    </node>
  
  </group>

</launch>
